### PR TITLE
fix: hanging call to close connection in the EFM plugin

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

Fix hanging call to close connection in the monitoring thread.

### Description

This PR fixes 3 issues:
1. when the monitoring thread calls connection.close() when using the MySQL driver, the MySQL driver attempts to obtain a lock on the Connection object. This object is already obtained by the main thread during Statement.executeQuery, thus causing a deadlock.
2. when the internal pool is used, the monitoring thread opens new connections using the HikariConnectionProvider.connect to establish new connections. When the monitoring thread opens connection using this provider, it would affect the internal connection pool by either attempting to reuse a connection from the pool or add a new connection to the pool. This leads to side effects such as a deadlocks.
3. The third issue is the reason this test hangs when using the MariaDB driver. When aborting the connection MariaDB creates a new StandardClient, which [creates a socket connection](https://github.com/mariadb-corporation/mariadb-connector-j/blob/d541afd6af963d931dd1a86adfb1cbcb1a0af10c/src/main/java/org/mariadb/jdbc/client/impl/StandardClient.java#L99), this call hangs because socketTimeout is 0. To fix this we need to populate the connection properties set in the test into the URL string in HikariPooledConnectionProvider before setting the jdbcUrl.

More details in https://github.com/orgs/awslabs/projects/91/views/13?pane=issue&itemId=24194630
Integration test result: https://github.com/Bit-Quill/aws-advanced-jdbc-wrapper/actions/runs/4759767214
### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by aaron-congo — APPROVED** [review]

File: wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorConnectionContext.java [hunk 1]
```diff
@@ -135,7 +136,7 @@ void abortConnection() {
     }
 
     try {
-      this.connectionToAbort.close();
+      this.connectionToAbort.abort(Executors.newSingleThreadExecutor());
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorConnectionContext.java:139]: What's the implications creating a new executor every time?

**Review by sergiyvamz — APPROVED** [review]

File: wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorConnectionContext.java (hunk 1 above)
Comment (karenc-bq) [reply wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorConnectionContext.java]: fixed

File: wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorConnectionContext.java [hunk 2]
```diff
@@ -29,6 +31,8 @@
 public class MonitorConnectionContext {
 
   private static final Logger LOGGER = Logger.getLogger(MonitorConnectionContext.class.getName());
+  private static final Executor ABORT_EXECUTOR = Executors.newSingleThreadExecutor();
+
```
Comment (karenc-bq) [inline wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorConnectionContext.java:35]:
```suggestion
```

File: wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java [hunk 3]
```diff
@@ -34,8 +34,10 @@
 import integration.refactored.TestInstanceInfo;
 import integration.refactored.container.ConnectionStringHelper;
 import integration.refactored.container.ProxyHelper;
+import integration.refactored.container.TestDriver;
```
Comment (karenc-bq) [inline wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java:37]:
```suggestion
```

File: wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java [hunk 4]
```diff
@@ -34,8 +34,10 @@
...
 import integration.refactored.container.TestDriverProvider;
 import integration.refactored.container.TestEnvironment;
+import integration.refactored.container.condition.DisableOnTestDriver;
```
Comment (karenc-bq) [inline wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java:40]:
```suggestion
```

**Pull request merged**